buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -803,7 +803,9 @@ class BeamModulePlugin implements Plugin<Project> {
 
       if (configuration.validateShadowJar) {
         project.task('validateShadedJarDoesntLeakNonOrgApacheBeamClasses', dependsOn: 'shadowJar') {
+          ext.outFile = project.file("${project.reportsDir}/${name}.out")
```

Comment:
Could you not have used a local variable with `def` instead of making it a public property on the project?

Ditto on the other one.

Reply:
Opened https://github.com/apache/beam/pull/6926

Reply:
Erm... I might to have a look again, but it is meant to be task.ext, i.e. a task extension property.

Of course, not a global project property.

Reply:
Should be, see https://docs.gradle.org/current/dsl/org.gradle.api.plugins.ExtensionAware.html

> Many Gradle objects are extension aware. This includes; projects, tasks, configurations, dependencies etc.